### PR TITLE
Add bundle extensions to auto-reloading watcher

### LINKED ISSUE #17694: env EXTENSIONS_AUTO_RELOAD not working with bundle extensions
### Checklist

- [X] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.
- [X] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).

### Describe the Bug

The env setting EXTENSIONS_AUTO_RELOAD is working fine with any other extension except bundle extensions.

### To Reproduce

Create a bundle extension and set `EXTENSIONS_AUTO_RELOAD=true` in your directus env settings. 

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by bourn404**
Adds paths for non-nested extensions to the chokidar function in order to enable auto reloading for bundle type extensions

Fixes #17694

### CODE REVIEW

**Comment by bourn404** [comment]
I put together this PR as a stopgap, under the assumption that the new system @Nitwel is putting together for extensions may still take some time to get merged.

File: api/src/extensions.ts [hunk 1]
```diff
@@ -294,9 +294,20 @@ class ExtensionManager {
 	private initializeWatcher(): void {
 		logger.info('Watching extensions for changes...');
 
-		const localExtensionPaths = NESTED_EXTENSION_TYPES.flatMap((type) => {
-			const typeDir = path.posix.join(pathToRelativeUrl(env['EXTENSIONS_PATH']), pluralize(type));
-			const fileExts = ['js', 'mjs', 'cjs'];
+		const extensionsPath = pathToRelativeUrl(env['EXTENSIONS_PATH']);
+		const fileExts = ['js', 'mjs', 'cjs'];
+
+		const rootExtensionDirs = path.posix.join(extensionsPath, '{directus-extension-*,@directus/extension-*}');
```
Comment (paescuj) [inline api/src/extensions.ts:300]: What's the idea behind `@directus/extension-*`?

Comment (bourn404) [reply api/src/extensions.ts:300]: I used the `EXTENSION_NAME_REGEX` from `/packages/constants/src/extensions` to know what the expected structure for extension names was.  That regex includes the `@directus/extension-*` pattern. https://github.com/directus/directus/blob/main/packages/constants/src/extensions.ts#L29

**Comment by rijkvanzanten** [comment]
Thanks @bourn404! While reviewing/researching this, @nickrum (original author of this logic) actually found a slightly simpler fix, so we'll apply that one instead (ref https://github.com/directus/directus/pull/18287). Thanks anyways! ❤️

**Pull request closed without merging**